consensus/core/src/core.rs

```diff
@@ -110,6 +103,26 @@ impl Core {
         // Try to accept them via the block manager
         let accepted_blocks = self.block_manager.add_blocks(blocks);
 
+        // Now process them, basically move the threshold clock and add them to pending list
+        self.add_accepted_blocks(accepted_blocks, None);
+
+        // Attempt to create a new block
+        let _ = self.try_new_block(false);
+
+        // TODO: we don't deal for now with missed references, will address later.
+        vec![]
+    }
+
+    /// Adds/processed all the newly `accepted_blocks`. We basically try to move the threshold clock and add them to the
+    /// pending ancestors list. The `pending_ancestors_retain_rounds` if defined then the method will retain on the pending ancestors
+    /// only the `pending_ancestors_retain_rounds` from the last formed quorum round. For example if set to zero (0), then
+    /// we'll strictly keep in the pending ancestors list the blocks of round >= last_quorum_round. If not defined, so None
+    /// is provided, then all the pending ancestors will be kep until the next block proposal.
+    fn add_accepted_blocks(
```

Comment:
what about `process_accepted_blocks`?

Reply:
to be honest I've kept the `add_` mostly for consistency with the rest of the API